a3fe/run/stage.py

```diff
@@ -1214,13 +1213,10 @@ def update(self, save_name: str = "output_saved") -> None:
             raise RuntimeError("Can't update while ensemble is running")
         if _os.path.isdir(self.output_dir):
             self._mv_output(save_name)
-        # Update the list of lambda windows in the simfile
-        _write_simfile_option(
-            simfile=f"{self.input_dir}/somd.cfg",
-            option="lambda array",
-            value=", ".join([str(lam) for lam in self.lam_vals]),
-        )
-        # Store the previous lambda window attributes that we want to preserve
+
+        self.engine_config.lambda_array = ", ".join([str(lam) for lam in self.lam_vals])
```

Comment:
Let's not store the lambda values in the engine config, but only pass when we're writing out the config, then we won't need to set this.